client/config/config.go

```diff
@@ -0,0 +1,96 @@
+package config
+
+import (
+	//	"fmt"
+	"path/filepath"
+
+	"github.com/cosmos/cosmos-sdk/client"
+)
+
+// Default constants
+const (
+	chainID        = ""
+	keyringBackend = "os"
+	output         = "text"
+	node           = "tcp://localhost:26657"
+	broadcastMode  = "sync"
+
+//	trace          = false
+)
+
+type ClientConfig struct {
+	ChainID        string `mapstructure:"chain-id" json:"chain-id"`
+	KeyringBackend string `mapstructure:"keyring-backend" json:"keyring-backend"`
+	Output         string `mapstructure:"output" json:"output"`
+	Node           string `mapstructure:"node" json:"node"`
+	BroadcastMode  string `mapstructure:"broadcast-mode" json:"broadcast-mode"`
+	//	Trace          bool   `mapstructure:"trace" json:"trace"`
+}
+
+func DefaultClientConfig() *ClientConfig {
+	return &ClientConfig{chainID, keyringBackend, output, node, broadcastMode}
+}
+
+func (c *ClientConfig) SetChainID(chainID string) {
+	c.ChainID = chainID
+}
+
+func (c *ClientConfig) SetKeyringBackend(keyringBackend string) {
+	c.KeyringBackend = keyringBackend
+}
+
+func (c *ClientConfig) SetOutput(output string) {
+	c.Output = output
+}
+
+func (c *ClientConfig) SetNode(node string) {
+	c.Node = node
+}
+
+func (c *ClientConfig) SetBroadcastMode(broadcastMode string) {
+	c.BroadcastMode = broadcastMode
+}
+
+/*
+func (c *ClientConfig) SetTrace(trace string) error {
+	boolVal, err := strconv.ParseBool(trace)
+	if err != nil {
+		return err
+	}
+	c.Trace = boolVal
+	return nil
+}
+*/
+
+func UpdateClientContextFromClientConfig(ctx client.Context) client.Context {
+	configPath := filepath.Join(ctx.HomeDir, "config")
+
+	/*
+		if err := ensureConfigPath(configPath); err != nil {
```

Comment:
In this function we need to keep this.

Moreover, if os.Stat(configFile) doesn't exist, we need to create it and populate it with DefaultClientConfig

Reply:
Yeah, sounds good